src/integration-test/helpers/clients/idamClient.ts

```diff
@@ -95,49 +92,32 @@ export class IdamClient {
    * @returns {Promise<string>}
    */
   static async upliftUser (email: string, upliftToken: string): Promise<void> {
-    if (strategicIdam) {
-      const upliftParams = IdamClient.toUrlParams({
-        userName: email,
-        password: defaultPassword,
-        jwt: upliftToken,
-        clientId: oauth2.client_id,
-        redirectUri: oauth2.redirect_uri
-      })
-
-      const res = await require('request-promise-native').post({
-        uri: `${baseURL}/login/uplift?${upliftParams}`,
-        headers: {
-          'Content-Type': 'application/x-www-form-urlencoded'
-        },
-        simple: false,
-        followRedirect: false,
-        json: false,
-        resolveWithFullResponse: true
-      })
-
-      const code: string = url.parse(res.headers.location, true).query.code.toString()
-      await IdamClient.exchangeCode(code)
-    } else {
-      const base64EncodedCredentials = IdamClient.toBase64(`${email}:${defaultPassword}`)
-      const upliftParams = IdamClient.toUrlParams({
-        response_type: 'code',
-        client_id: oauth2.client_id,
-        redirect_uri: oauth2.redirect_uri,
-        upliftToken: upliftToken
-      })
-
-      const options = {
-        method: 'POST',
-        uri: `${baseURL}/oauth2/authorize?${upliftParams}`,
-        headers: {
-          Authorization: `Basic ${base64EncodedCredentials}`
-        }
-      }
-      return request(options).then(function (response) {
-        return Promise.resolve()
-      })
+    let code
```

Comment:
does this need to be set here? Can't it be a const inside the .then() ?

Reply:
yeah - i tried this but kept getting tslint errors:
```
error TS2322: Type 'string | string[]' is not assignable to type 'string'.
  Type 'string[]' is not assignable to type 'string'.
```
🤔 have changed to: `const code: any = url.parse(response.headers.location, true).query.code`